app/assets/javascripts/govuk_publishing_components/components/intervention.js

```diff
@@ -0,0 +1,80 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  function Intervention ($module) {
+    this.$module = $module
+    this.$banner = this.$module.querySelector('.gem-c-intervention')
+    this.$closeLink = this.$module.querySelector('.gem-c-intervention__dismiss-link')
+    this.$campaignName = this.$module.getAttribute('data-intervention-name')
+    this.$interventionCookie = window.GOVUK.cookie('intervention_campaign')
+  }
+
+  Intervention.prototype.init = function () {
+    this.$module.close = this.handleClose.bind(this)
+
+    if (this.$closeLink) {
+      this.$closeLink.addEventListener('click', this.$module.close)
+    }
+
+    if (this.cookieHasCampaign()) {
+      this.hideBanner()
+    }
+  }
+
+  Intervention.prototype.handleClose = function (event) {
+    if (event) {
+      event.preventDefault()
+    }
+
+    if (this.cookieHasCampaign()) {
+      this.setCookies()
+    }
+    this.hideBanner()
+  }
+
+  Intervention.prototype.cookieHasCampaign = function () {
+    if (!this.$interventionCookie) {
+      return false
+    } else {
+      var included = this.cookieValues().includes(this.$campaignName)
+      return included
+    }
+  }
+
+  Intervention.prototype.readCampaignCookie = function () {
+    return window.GOVUK.cookie('intervention_campaign') || ''
+  }
+
+  Intervention.prototype.cookieValues = function () {
+    var cookieString = this.readCampaignCookie()
+
+    if (cookieString !== null && cookieString.length > 0) {
+      return cookieString.split(',')
+    }
+  }
+
+  Intervention.prototype.appendCookieValues = function () {
+    var commaPrefixed = ','.concat(this.$campaignName)
+    return this.readCampaignCookie().concat(commaPrefixed)
```

Comment:
Interested by the use of `concat` here as apparently it's intended for merging arrays, which I don't think any of these things are?

I think this function could be simplified to

```JavaScript
return this.readCampaignCookie() + ',' + this.$campaignName
```

Reply:
It is a String function https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/concat 

and supported as far back as IE 

https://caniuse.com/mdn-javascript_builtins_string_concat

and much easier to read than the + operator

Reply:
but maybe it can be simplified...